samples/memcached-api-server/test.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"log"
+	"runtime"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+func main() {
```

Comment:
Should this be a main or a `go test`?  e.g. do we want to fail with Fatal or with `Fail` / `FailNow`

Reply:
In this level of testing, some commands must be executed before others. For example, `generate` must be run before `go build ./pkg/...`, `create docs` must be run before `diff docs`. If the prior one failed, the execution should exit. 

I should restructure the testing so that each execution is only for one kubebuilder command. The failure from one execunion won't hide other testings. Having a package like `internal/test/e2e/pkg` will help this.

Reply:
I think it will be idiomatic to have it as `go test`. 

@Liujingfang1 in order to execute serial execution of dependencies, we can use `TestMain` https://golang.org/pkg/testing/#hdr-Main